drydock/plugin.py

```diff
@@ -1,11 +1,140 @@
 from glob import glob
 import os
+import click
 import pkg_resources
 
-from tutor import hooks
+import typing as t
+
+import functools
+from .hooks import SYNC_WAVES_ORDER_ATTRS_TYPE, SYNC_WAVES_ORDER
+
+from tutor import hooks as tutor_hooks
+from tutor import env as tutor_env
+from tutor import serialize, types
+from tutor import config as tutor_config
 
 from .__about__ import __version__
 
+INIT_JOBS_SYNC_WAVE = 1
+
+# This function is taken from
+# https://github.com/overhangio/tutor/blob/v15.3.7/tutor/commands/k8s.py#L180
+def _load_jobs(tutor_conf: types.Config) -> t.Iterable[t.Any]:
+    jobs = tutor_env.render_file(tutor_conf, "k8s", "jobs.yml").strip()
+    for manifest in serialize.load_all(jobs):
+        if manifest["kind"] == "Job":
+            yield manifest
+
+
+# The definition of the init tasks is taken and adapted from
+# https://github.com/overhangio/tutor/blob/v15.3.7/tutor/commands/jobs.py#L64
+# and https://github.com/overhangio/tutor/blob/v15.3.7/tutor/commands/k8s.py#L80
+def get_init_tasks():
+    """Return the list of init tasks to run."""
+    init_tasks = list(tutor_hooks.Filters.CLI_DO_INIT_TASKS.iterate())
+    context = click.get_current_context().obj
+    tutor_conf = tutor_config.load(context.root)
+
+    # Standarize deprecated COMMANDS_INIT and COMMANDS_PRE_INIT Filter
+    pre_init_tasks = []
+    for service, init_path in tutor_hooks.Filters.COMMANDS_PRE_INIT.iterate():
+        pre_init_tasks.append((service, tutor_env.read_template_file(*init_path)))
+
+    init_tasks = pre_init_tasks + init_tasks
```

Comment:
can we start building the list from the pre_init tasks and extend at the end? to avoid the intermediate lists?

```python
init_tasks = []
for ...:
  init_tasks.append(...)
...
init_tasks.extend(tutor_hooks.Filters.CLI_DO_INIT_TASKS.iterate())
```